metadata-ingestion/src/datahub/ingestion/source/hex/datahub.py

```diff
@@ -0,0 +1,385 @@
+                    logger.debug(
+                        f"Skipping query {query_urn} - failed to extract Hex metadata"
+                    )
+                    self.report.filtered_out_queries_missing_metadata += 1
+                    continue
+
+                hex_project_id, workspace_from_url = metadata_result
+
+                # Validate workspace
+                if workspace_from_url != self.workspace_name:
+                    logger.debug(
+                        f"Skipping query {query_urn} - workspace '{workspace_from_url}' doesn't match '{self.workspace_name}'"
+                    )
+                    self.report.filtered_out_queries_different_workspace += 1
+                    continue
+
+                # Extract subjects
+                dataset_subjects: List[DatasetUrn] = []
+                schema_field_subjects: List[SchemaFieldUrn] = []
+                for subject in aspects.querySubjects.value.subjects:
+                    if subject.entity and subject.entity.startswith("urn:li:dataset:"):
+                        dataset_subjects.append(DatasetUrn.from_string(subject.entity))
+                    elif subject.entity and subject.entity.startswith(
+                        "urn:li:schemaField:"
+                    ):
+                        schema_field_subjects.append(
+                            SchemaFieldUrn.from_string(subject.entity)
+                        )
+
+                if not dataset_subjects and not schema_field_subjects:
+                    self.report.filtered_out_queries_no_subjects += 1
+                    continue
+
+                # Create response
+                response = QueryResponse(
+                    urn=query_urn,
+                    hex_project_id=hex_project_id,
+                    dataset_subjects=dataset_subjects,
+                    schema_field_subjects=schema_field_subjects,
+                )
+                logger.debug(
+                    f"Succesfully extracted {len(dataset_subjects)} dataset subjects and {len(schema_field_subjects)} schema field subjects for query {query_urn}: {dataset_subjects} {schema_field_subjects}"
+                )
+                self.report.total_queries += 1
+                self.report.total_dataset_subjects += len(dataset_subjects)
+                self.report.total_schema_field_subjects += len(schema_field_subjects)
+
+                logger.debug(
+                    f"Processed query {query_urn} with Hex project ID {hex_project_id}"
+                )
+                yield response
+
+    def _fetch_query_entities(self, query_urns: List[QueryUrn]) -> EntitiesResponse:
+        entities_by_urn = EntitiesResponse()
+
+        for i in range(0, len(query_urns), self.page_size):
+            batch = query_urns[i : i + self.page_size]
+
+            logger.debug(f"Fetching query entities for {len(batch)} queries: {batch}")
+            response_json = self.datahub_client._graph.get_entities_v2(
```

Comment:
I think we actually should be using the openapi v3 endpoints. If we used those, we should be able to do things like `QueryPropertiesClass.from_obj(...)` - redefining the models that we already have from codegen feels crazy

Would recommend chatting with @david-leifker about this - ideally we'd just add this as a utility method to the DataHubGraph class

Reply:
I just checked `/openapi/v3/entity/query/batchGet` and yes, it is modelled with the datahub model classes... this is 🔝 !

<img width="1592" alt="Screenshot 2025-03-31 at 13 28 13" src="https://github.com/user-attachments/assets/80c19a99-f6cf-459a-8cbf-e3b17e375a7d" />

Reply:
Also, I found that I can specify for each URN the fields I want to fetch

```
curl -X 'POST' \
  'https://dev04.acryl.io/openapi/v3/entity/query/batchGet?systemMetadata=false' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '[
  {
    "urn": "urn:li:query:00f7b2c3fd28356cd6afe884176df9d85ae25b138fa143cdf85b34c3b24b4a68",
    "queryProperties" :{}
  },

  {
    "urn": "urn:li:query:0165bc1ae0bef3b336fa9fea24e31f2ae6f45bdd8c21f8fb59c8f682be5fed30",
    "queryProperties" :{}
  },

  {
    "urn": "urn:li:query:040931b3b0da2fba45ff475e18f775429dc37ab0cc928afe34af5501814d0de9",
    "queryProperties" :{},
    "querySubjects": {}
  }
]'
```

🥇

Reply:
Moved to use the new `DataHubGraph#get_entities` method (which is based on OpenAPI v3 endpoints) in commit f61593fc1b358c62ed5ba52ad210e358abb4c6a3
As a side effect, all the pydantic models I had defined exclusively for the deserialization are gone!